tests/integration/test_duckdb_tpch.py

```diff
@@ -0,0 +1,95 @@
+import duckdb
+import pytest
+
+from ..common import SubstraitUtils
+from ..consumers.duckdb_consumer import DuckDBConsumer
+from ..parametrization import custom_parametrization
+from ..verification import verify_equals
+from .queries.tpch_test_cases import TPCH_QUERY_TESTS
+
+
+@pytest.mark.usefixtures("prepare_tpch_parquet_data")
+class TestDuckDBConsumer:
+    """
+    Test Class for testing Substrait using Acero as a consumer.
+    """
+
+    @staticmethod
+    @pytest.fixture(scope="class", autouse=True)
+    def setup_teardown_class(request):
+        cls = request.cls
+
+        cls.db_connection = duckdb.connect()
+        cls.db_connection.execute("INSTALL substrait")
+        cls.db_connection.execute("LOAD substrait")
+        cls.consumer = DuckDBConsumer()
+        cls.utils = SubstraitUtils()
+        cls.created_tables = set()
+
+        yield
+
+        cls.db_connection.close()
+
+    @custom_parametrization(TPCH_QUERY_TESTS)
+    def test_substrait_query(
+        self,
+        test_name: str,
+        file_names: list,
+        sql_query: str,
+        substrait_query: str,
+        sort_results: bool = False,
+    ) -> None:
+        """
+        1.  Load all the parquet files into DuckDB as separate tables.
+        2.  Format the SQL query to work with DuckDB by inserting all the table names.
+        3.  Execute the SQL on DuckDB.
+        4.  Run the substrait query plan.
+        5.  Compare substrait query plan results against the results of
+            running the SQL on DuckDB.
+
+        Parameters:
+            test_name:
+                Name of test.
+            file_names:
+                List of parquet files.
+            sql_query:
+                SQL query.
+        """
+
+        parquet_file_paths = self.utils.get_full_path(file_names)
```

Comment:
Since you have this utility, maybe it should be rolled into the `consumer` class?  That way you only have to pass in the `file_names` (and then have the `consumer` class figure out the paths using your helper method)

Reply:
Good idea.  Done.